src/merkle/path_set.rs

```diff
@@ -73,7 +73,7 @@ impl MerklePathSet {
         let path_key = index.value() - parity;
         self.paths
             .get(&path_key)
-            .ok_or(MerkleError::NodeNotInSet(path_key))
+            .ok_or(MerkleError::NodeNotInSet(index))
```

Comment:
Was the use of `path_key` a bug?

Reply:
Kind of. `MerklePathSet` stores data for both even and odd sibling leaves at the even sibling's location. So, if one doesn't exist the other one won't either. In this case, we are checking whether the even sibling is in the tree - so, technically this check only checks for even siblings. But a more appropriate error message would be to use the requested index.